bufferserver/src/main/java/com/datatorrent/bufferserver/internal/LogicalNode.java

```diff
@@ -220,17 +220,17 @@ public void catchUp()
       }
 
       if (iterator.hasNext()) {
-        addedData();
+        addedData(false);
       }
     }
 
     logger.debug("Exiting catch up because caughtup = {}", caughtup);
   }
 
   @Override
-  public boolean addedData()
+  public boolean addedData(boolean checkIfListenerHaveDataToSendOnly)
   {
-    if (isReady()) {
+    if (!checkIfListenerHaveDataToSendOnly && isReady()) {
```

Comment:
It is a little weird that to only check if data is available, one would need to go through the method that is trying to send data. What if in future there are other things to do when subscriber is not ready, are they likely to apply to the check case as well?